src/styles/main.scss

```diff
@@ -1,3 +1,56 @@
-body {
-  margin: 0;
+@import 'variables';
+@import url('./reset.scss');
+
+.calendar {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: $padding;
+  width: ($padding * 2) + ($day-size * 7) + ($gap * 6);
+
+  @each $day, $left-margin in $days {
+    &--start-day--#{$day} .calendar__day:first-child {
+      margin-left: $left-margin;
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &--month-length--#{$index} .calendar__day:nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    width: $day-size;
+    height: $day-size;
+    border: 1px solid black;
+    background-color: #eee;
+    font-family: Arial, sans-serif;
+    position: relative;
+
+    transition: all 0.5;
```

Comment:
to improve optimization it's better to specify properties you change on hover

Fix all such cases